### PR TITLE
postgres API tests

### PR DESCRIPTION
**Pull request by jerith**

### CODE REVIEW

**Comment by miltontony** [comment]
🍰 for re-organising

File: .travis.yml [hunk 1]
```diff
@@ -1,9 +1,35 @@
...
+    - $HOME/.pyenv_cache
 services:
-    - redis-server
+  - postgresql
+
+before_install:
+  - psql -c 'create database xylem;' -U postgres
+  - |
+      if [[ -n "$PYPY_VERSION" ]]; then
+        wget https://github.com/praekeltfoundation/travis-pypy/releases/download/0.1.0/setup-pypy.sh
```
Comment (JayH5) [inline .travis.yml]: I did a travis-pyenv release finally yesterday. Check the guide: https://github.com/praekeltfoundation/travis-pyenv#example-travisyml

Comment (jerith) [reply .travis.yml]: I copied this directly from `marathon-acme`, and we're not using pypy yet anyway. I'll update for this anyway, thanks.

**Review by JayH5 — APPROVED** [review]
:+1:

**Review by JayH5 — APPROVED** [review]
:+1:

**Pull request merged**